uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/document-list.controller.spec.js

```diff
@@ -344,6 +341,107 @@ define([
       });
     });
 
+    describe('date range filter', function () {
+      var sampleDate1 = '14/04/1989';
+      var sampleDate2 = '17/08/1989';
+
+      beforeEach(function () {
+        initController();
+      });
+
+      describe('when the "from" date is changed', function () {
+        beforeEach(function () {
+          controller.filterParamsHolder.dateRange.from = sampleDate1;
+
+          $rootScope.$digest();
+        });
+
+        it('sets the minimum "to" date', function () {
+          expect(controller.datepickerOptions.until.minDate).toBe(sampleDate1);
+        });
+      });
+
+      describe('when the "to" date is changed', function () {
+        beforeEach(function () {
+          controller.filterParamsHolder.dateRange.until = sampleDate2;
+
+          $rootScope.$digest();
+        });
+
+        it('sets the maximum "from" date', function () {
+          expect(controller.datepickerOptions.from.maxDate).toBe(sampleDate2);
+        });
+      });
+    });
+
+    describe('filter tabs', function () {
+      var allDocumentStatuses = _.pluck(documentFabricator.documentStatus(), 'key');
+      var onlyPendingDocumentStatuses = getDocumentStatusesIds(
+        ['awaiting upload', 'awaiting approval']);
+      var sampleDocumentStatuses = getDocumentStatusesIds(
+        ['awaiting upload', 'approved', 'rejected']);
+      var tabsShowingOnlyPendingDocuments = [
+        { title: 'Overdue', value: 'overdue' },
+        { title: 'Next fortnight', value: 'dueInNextFortnight' },
+        { title: '90 days', value: 'dueInNinetyDays' }
+      ];
+
+      beforeEach(function () {
+        initController();
+        fabricateDocumentStatuses();
+
+        controller.filterParamsHolder.documentStatus = sampleDocumentStatuses;
+      });
+
+      tabsShowingOnlyPendingDocuments.forEach(function (tab) {
+        describe('user selects "' + tab.title + '" filter tab', function () {
```

Comment:
Can we use `when the user selects ...` ? In that way it looks like it's telling a story.

Reply:
Done.